### PR TITLE
Jenkins Pipeline Git Commimt

### PR DESCRIPTION
**Pull request by gilbode**
Jenkins pipeline makes a git commit at the end of the build & if there were subsequent commits since the build started git push fails.  To make this failure less likely move the git commit to beginning of Jenkinsfile.  In the event the commit fails, uses the groovy postbuild plugin to update build description to describe why it failed.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -99,6 +99,46 @@ pipeline {
...
+                    git checkout -b ${env.BRANCH_NAME}
+                    git add deploy/operator.yaml
+                    git commit -m "[verrazzano] Update verrazzano-platform-operator image version to ${DOCKER_IMAGE_TAG} in operator.yaml"
+                    git push origin ${env.BRANCH_NAME}
+                   """
+            }
+            post {
+                unsuccessful {
+                    script {
+                        currentBuild.description = "Git commit of operator.yaml failed"
```
Comment (markxnelson) [inline Jenkinsfile:136]: could we add an instruction to retry/rerun?

**Review by markxnelson — DISMISSED** [review]
one small request, but approving anyway

File: Jenkinsfile (hunk 1 above)
Comment (gilbode) [reply Jenkinsfile:136]: i thought about that but worried it would be confusing if you didn't read the message in real time.

File: Jenkinsfile [hunk 2]
```diff
@@ -99,6 +99,46 @@ pipeline {
             }
         }
 
+        stage('Generate operator.yaml') {
```
Comment (mgianatagh) [inline Jenkinsfile:102]: should we combine these into a single stage?

Comment (gilbode) [reply Jenkinsfile:102]: i think generate operator.yaml might be needed on all branches

**Pull request merged**